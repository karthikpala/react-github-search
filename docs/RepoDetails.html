<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: RepoDetails</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: RepoDetails</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>RepoDetails<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">Component to render and display RepoDetails details like full name, avatar, description, license etc - styled as a card</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="RepoDetails"><span class="type-signature"></span>new RepoDetails<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Create a RepoDetails component
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    
    <dt class="tag-author">Author:</dt>
    <dd class="tag-author">
        <ul>
            <li>[Karthik Pala]</li>
        </ul>
    </dd>
    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_RepoDetails_RepoDetails.jsx.html">components/RepoDetails/RepoDetails.jsx</a>, <a href="components_RepoDetails_RepoDetails.jsx.html#line13">line 13</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    
        <h3 class="subsection-title">Extends</h3>

        


    <ul>
        <li>PureComponent</li>
    </ul>


    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="render"><span class="type-signature"></span>render<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    React render Method
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_RepoDetails_RepoDetails.jsx.html">components/RepoDetails/RepoDetails.jsx</a>, <a href="components_RepoDetails_RepoDetails.jsx.html#line20">line 20</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="APIServices.html">APIServices</a></li><li><a href="App.html">App</a></li><li><a href="RepoDetails.html">RepoDetails</a></li><li><a href="RepoList.html">RepoList</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.3</a> on Wed Jun 02 2021 21:32:02 GMT+0530 (India Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>