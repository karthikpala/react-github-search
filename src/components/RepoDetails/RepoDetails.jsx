import React, { PureComponent } from 'react';
import { object } from 'prop-types';

/**
 * Component to render and display RepoDetails details like full name, avatar, description, license etc - styled as a card
 * @author [Karthik Pala]
 * @extends PureComponent
 */
class RepoDetails extends PureComponent {
  /**
   * Create a RepoDetails component
   */
  constructor() {
    super();
  }

  /**
   * React render Method
   */
  render() {
    return (
      <>
        <div className="card__avatar">
          <img
            className="card__avatar__img"
            src={this.props.item && this.props.item.owner.avatar_url}
            draggable="false"
          ></img>
        </div>
        <div className="card__name">
          <h2>{this.props.item.full_name}</h2>
        </div>

        {this.props.item.description && (
          <div className="flex">
            <span className="card__text">
              <span className="card__label">Description</span>
              {this.props.item.description}
            </span>
          </div>
        )}
        
        {this.props.item.forks !==undefined && (
          <div className="flex">
            <span className="card__text">
              <span className="card__label">Forks</span>
              {this.props.item.forks}
            </span>
          </div>
        )}

        {this.props.item.watchers !== undefined && (
          <div className="flex">
            <span className="card__text">
              <span className="card__label">Watchers</span>
              {this.props.item.watchers}
            </span>
          </div>
        )}
        
        {this.props.item.open_issues !== undefined && (
          <div className="flex">
            <span className="card__text">
              <span className="card__label">Open Issues</span>
              {this.props.item.open_issues}
            </span>
          </div>
        )}

        {this.props.item.license?.name && (
          <div className="flex">
            <span className="card__text">
              <span className="card__label">License</span>
              {this.props.item.license.name}
            </span>
          </div>
        )}

        {this.props.item.html_url && (
          <div className="flex" onClick={() => {window.open(this.props.item.html_url, "_blank")}}>
            <span className="card__text">
              <i className="fa fa-arrow-circle-right card__icon"></i>
            </span>
          </div>
        )}
      </>
    );
  }
}

RepoDetails.propTypes = {
  item: object.isRequired,
};

export default RepoDetails;
