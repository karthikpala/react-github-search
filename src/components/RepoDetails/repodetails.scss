.card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  word-break: break-word;
  text-align: center;
  div{
    margin: 5px auto;
  }

  &__avatar {
    width: 120px;
    height: 120px;
    position: relative;

    &__img {
      border-radius: 50%;
      min-height: 100%;
      max-width: 100%;
      border: 2px solid $primary-color;
    }
  }

  &__name::after {
    content: "";
    background-color: $primary-color;
    height: 1px;
    width: 100px;
    margin: 0 auto 0 auto;
    display: block;
  }

  &__text {
    text-align: center;
    color: $neutral-light;
    font-weight: bold;
  }

  &__label {
    font-weight: bold;

    &:after {
      content: ' : ';
    }
  }

  &__icon {
    margin-right: 5px;
    color: $neutral-dark;
  }
}

.card--theme-primary {
  .card__avatar__img {
    border: 2px solid white;
  }

  .card__text {
    font-weight: 500;
    color: white
  }

  .card__icon {
    color: white;
    font-size: 50px;
    cursor: pointer;
  }
}