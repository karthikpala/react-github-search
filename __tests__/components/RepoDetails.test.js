import React from 'react';
import RepoDetails from '../../src/components/RepoDetails/RepoDetails';
import repoResult from '../../__mocks__/repoResult.json';
import { create } from 'react-test-renderer';

describe('RepoDetails', () => {
  it('should render RepoDetails component correctly', () => {
    const item = repoResult.items[0];
    const component = create(<RepoDetails item={item} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
