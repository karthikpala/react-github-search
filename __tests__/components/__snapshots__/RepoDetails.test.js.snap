// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoDetails should render RepoDetails component correctly 1`] = `
Array [
  <div
    className="ucard__avatar"
  >
    <img
      className="ucard__avatar__img"
      draggable="false"
      src="https://avatars.githubusercontent.com/u/80134?v=4"
    />
  </div>,
  <div
    className="ucard__name"
  >
    <h2>
      jfcmacro/ewe
    </h2>
  </div>,
  <div
    className="flex"
  >
    <span
      className="ucard__text"
    >
      <span
        className="ucard__label"
      >
        Description
      </span>
      An Haskell Implementation of the EWE (Kent) Programming Language, an extension of the RAM Language (Sethi).
    </span>
  </div>,
  <div
    className="flex"
  >
    <span
      className="ucard__text"
    >
      <span
        className="ucard__label"
      >
        Forks
      </span>
      3
    </span>
  </div>,
  <div
    className="flex"
  >
    <span
      className="ucard__text"
    >
      <span
        className="ucard__label"
      >
        Watchers
      </span>
      7
    </span>
  </div>,
  <div
    className="flex"
  >
    <span
      className="ucard__text"
    >
      <span
        className="ucard__label"
      >
        Open Issues
      </span>
      0
    </span>
  </div>,
  <div
    className="flex"
  >
    <span
      className="ucard__text"
    >
      <span
        className="ucard__label"
      >
        License
      </span>
      Other
    </span>
  </div>,
  <div
    className="flex"
    onClick={[Function]}
  >
    <span
      className="ucard__text"
    >
      <i
        className="fa fa-arrow-circle-right ucard__icon"
      />
    </span>
  </div>,
]
`;
